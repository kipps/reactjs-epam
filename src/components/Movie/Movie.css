.Movie {

}

img {
    width: 100%;
}
