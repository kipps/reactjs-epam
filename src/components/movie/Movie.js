import React from 'react';
import PropTypes from 'prop-types';
import {Link} from 'react-router-dom';

const Movie = (props) => {
  const path = (window.location.pathname === '/' || window.location.pathname.includes('/search/')) ? `/film/${props.id}` : props.id;

  return (
    <div className={'Movie mb-32'}>
      <a href={`${path}`} className='link-clear block'>
        <img alt={props.title} src={props.poster_path} className='Movie__img'/>
      </a>
      <div>
        <div className='mb-8'>
          <div className='Movie__date mb-8'>
            {props.release_date}
          </div>
          <h3>
            <a href={`${path}`} className='link-clear block'>{props.title}</a></h3>
        </div>
        <div className='Movie__genres'>
          <span>
            {props.genres.map((g, i) => <small key={i}>{g}</small>)}
         </span>
        </div>
      </div>
    </div>
  );
}

Movie.propTypes = {
  id: PropTypes.number,
  title: PropTypes.string,
  tagline: PropTypes.string,
  vote_average: PropTypes.number,
  vote_count: PropTypes.number,
  release_date: PropTypes.string,
  poster_path: PropTypes.string,
  overview: PropTypes.string,
  budget: PropTypes.number,
  revenue: PropTypes.number,
  runtime: PropTypes.number,
  genres: PropTypes.array
}

export default Movie;