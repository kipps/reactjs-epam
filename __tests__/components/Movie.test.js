import { shallow } from 'enzyme/build';
import { configure } from 'enzyme/build';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16/build';

import Movie from '../../src/components/movie/Movie';

configure({ adapter: new Adapter() });

describe("Movie", () => {
    it("renders default movie", () =>
        expect(
            shallow(<Movie />)
                .find('div.Movie')
                .length
        ).toBe(1));

    it('renders correctly', () => {
        const tree = renderer
          .create(<Movie />)
          .toJSON();
        expect(tree).toMatchSnapshot();
    });
});