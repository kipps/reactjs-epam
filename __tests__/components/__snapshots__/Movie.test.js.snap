// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie renders correctly 1`] = `
<div
  className="Movie"
>
  <img
    className="Movie__img"
    src="http://localhost/src/images/undefined.jpg"
  />
  <div>
    <div
      className="flex space-between v-center mb-8"
    >
      <h3 />
      <span
        className="Movie__date"
      />
    </div>
    <span>
      <small />
    </span>
  </div>
</div>
`;
